sigoptlite/builders.py

```diff
@@ -202,64 +21,14 @@ def create_experiment_from_template(experiment_template, **kwargs):
   return LocalExperimentBuilder(experiment_meta)
 
 
-def get_path_string(path):
-  strings = (f"[{part}]" if is_integer(part) else f".{part}" for part in path)
-  return "".join(strings)
-
-
-def is_integer(num):
-  if isinstance(num, bool):
-    return False
-  elif isinstance(num, Integral):
-    return True
-  else:
-    return False
-
-
-def process_error(e, class_name):
-  if e.validator == "type":
-    expected_type = str(e.schema["type"])
-    raise ValueError(f"Invalid type for {get_path_string(e.path)}, expected type {expected_type}")
-  elif e.validator in ["maxProperties", "minProperties"]:
-    least_most = "at least" if e.validator == "minProperties" else "at most"
-    raise ValueError(f"Expected {least_most} {e.validator_value} keys in {class_name}: {json.dumps(e.instance)}")
-  elif e.validator == "required":
-    if isinstance(e.instance, Mapping):
-      missing_keys = [key for key in e.validator_value if key not in e.instance]
-      missing_key = missing_keys[0] if len(missing_keys) > 0 else None
-    else:
-      missing_key = e.validator_value[0]
-    raise ValueError(f"Missing required json key `{missing_key}` in {class_name}: {json.dumps(e.instance)}")
-  elif e.validator in ["minimum", "maximum"]:
-    key = get_path_string(e.path)
-    greater_less = "greater than" if e.validator == "minimum" else "less than"
-    raise ValueError(f"{key} must be {greater_less} or equal to {e.validator_value}")
-  elif e.validator == "exclusiveMinimum":
-    key = get_path_string(e.path)
-    raise ValueError(f"{key} must be greater than {e.validator_value}")
-  elif e.validator in ["minLength", "maxLength", "minItems", "maxItems"]:
-    key = get_path_string(e.path)
-    greater_less = "greater than" if e.validator in ["minLength", "minItems"] else "less than"
-    raise ValueError(f"The length of {key} must be {greater_less} or equal to {e.validator_value}")
-  elif e.validator == "enum":
-    allowed_values = ", ".join([str(s) for s in e.validator_value if s is not None])
-    raise ValueError(f"{e.instance} is not one of the allowed values: {allowed_values}")
-  elif e.validator == "pattern":
-    raise ValueError(f"{e.instance} does not match the regular expression /{e.validator_value}/")
-  elif e.validator in ["oneOf", "anyOf"]:
-    if len(e.context) > 0:
-      process_error(e.context[0], class_name)
-    raise NotImplementedError("Error has no context but it is oneOf or anyOf related.")
-  else:
-    raise NotImplementedError(f"Unrecognized error {e.validator} parsing json {class_name}: {json.dumps(e.instance)}")
-
-
 class BuilderBase(object):
   def __new__(cls, input_dict, **kwargs):
     try:
       cls.validate_input_dict(input_dict)
     except AssertionError as e:
       raise ValueError(f"Invalid input for {cls.__name__} {e}") from e
+    except SigoptValidationError as e:
+      raise ValueError(f"Validation failed for {cls.__name__}: {e}") from e
```

Comment:
Prefer not outputting the names of internal objects  (like `LocalExperimentBuilder`) to the user. 

I think just directly raising SigoptValidationError is sufficient here.

Reply:
maybe this should be `cls.cls_name` instead of `cls.__name__`?  

cc @malkomes , maybe line 29 should also be `cls.cls_name`?

Reply:
At present, cls_name is only valid for Experiments, not any of the other entities that can throw an exception, but I can fix that and make it true for all builders, in which case it would be useful in this error message.